### PR TITLE
fix: fix nfts displayed on account and refactor collectibles component

### LINKED ISSUE #9196: [Bug]: NFTs owned by imported account is shown on wallet first account after importing
### Describe the bug

NFTs owned by an imported account (oragefox.eth) are showing for the first wallet account.  This is happening for both upgrading app or new installs.

Before upgrading from v7.19.0 to v7.20.0 my wallet had 3 wallet accounts, 1 imported account (orangefox.the), and 2 hw accounts.  The imported account owned multiple NFTs on Ethereum Mainnet and all other accounts did not own any NFTs on Ethereum Mainnet.  NFTs were properly being shown only for the imported account on the NFTs tab.

After upgrading to v7.20.0 I now see some NFTs owned by the imported account showing on the first wallet account (Account 1).  

The issue does not occur instantly after importing or upgrading.  Issue started showing soon after importing and then switching accounts.

This is also happening for new installs and importing an account that owns NFTs.
This is also happening for both iOS and Android devices.

Not able to reproduce on v7.19.0.

### Expected behavior

NFTs should only be displayed for the selected account that owns NFTs.

### Screenshots/Recordings

https://github.com/MetaMask/metamask-mobile/assets/6626407/05fb9705-0513-415c-94fc-456b3d9ef21d

### Steps to reproduce

GIVEN I installed MMM
AND I have completed onboarding
AND wallet is backed up
AND Ethereum Mainnet is selected
WHEN I import an account that owns NFTs
AND I enable Autodetect NFTs
THEN NFTs are displayed on NFTs view in main wallet view for imported account
WHEN I switch to the wallets first account
AND first account does not own NFTs
THEN no NFTs are displayed for first account (FAIL: NFTs from imported account are shown for first account)
AND NFTs are not displayed for any other account that doesn't own NFTs

### Error messages or log output

_No response_

### Version

7.20.0

### Build type

None

### Device

Any

### Operating system

iOS, Android

### Additional context

_No response_

### Severity

Sev1

### PR DESCRIPTION
**Pull request by sahar-fehri**
## **Description**

This PR fixes this issue https://github.com/MetaMask/metamask-mobile/issues/9196.
Patch created considering this open PR  on core https://github.com/MetaMask/core/pull/4143

When a user imports an account with Nfts then switches back and forth betwen two accounts, at some point he will see nfts from one account appear on another.

Link to core branch for the patch : https://github.com/MetaMask/core/compare/patch/mobile-assets-controllers-v-12-0-0...patch/mobile-assets-controllers-v-12-0-0-nft-bug?expand=1

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-mobile/issues/9196.
Core PR: https://github.com/MetaMask/core/pull/4143

## **Manual testing steps**

1. Open app
2. Import an account that has nfts (use for example orangefox.eth)
3. Create a new account (You can either create a new account which will be empty or use any account that you already have)
4. Switch back and forth between orangefox.eth and another account fast and you will see nfts from orangefox.eth that appear on the other account. (This should not happen on this PR)

### **Before**

https://github.com/MetaMask/metamask-mobile/assets/10994169/611bd2c5-b005-48ed-9926-6d72ff244816

### **After**

Switching back and forth between accounts

https://github.com/MetaMask/metamask-mobile/assets/10994169/5c00c78d-f3f2-4080-b441-eaf9828aec13

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: patches/@metamask+assets-controllers+12.0.0.patch [hunk 1]
```diff
@@ -2861,8 +2936,6 @@ index 0000000..2e07fa7
 +    '0x4e454152',
 +    // Harmony Mainnet Shard 0
 +    '0x63564c40',
-+    // Linea Mainnet
-+    '0xe708'
```
Comment (tommasini) [inline patches/@metamask+assets-controllers+12.0.0.patch:2865]: Something went wrong, I think we do not want to delete linea mainnet from here, do we want?

**Review by tommasini — DISMISSED** [review]

File: patches/@metamask+assets-controllers+12.0.0.patch (hunk 1 above)
Comment (sahar-fehri) [reply patches/@metamask+assets-controllers+12.0.0.patch:2865]: Hmmm not sure what happened here, that file was created automatically, i only updated the fct that i need in nftController.ts on core then did yarn build, then on mobile `yarn patch:assets ../core`

**Comment by chris6090** [comment]
> ## [https://bitrise.io/](https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/fc980f77-c265-4d20-8fde-301160e3648c) **Bitrise**
> 
> ✅✅✅ `pr_smoke_e2e_pipeline` passed on Bitrise! ✅✅✅
> 
> Commit hash: cf447a645e6d719f1d22d22063babc8878924620
> Build link: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/fc980f77-c265-4d20-8fde-301160e3648c
> 
> >[!NOTE]
> >- You can kick off another `pr_smoke_e2e_pipeline` on Bitrise by removing and re-applying the `Run Smoke E2E` label on the pull request
> 
> 
> 
> 
> 

kriseng83@gmail.com

**Comment by tommasini** [comment]
Also, can you update this core branch with these changes to run the script from it?
`patch/mobile-assets-controllers-v-12-0-0`

**Review by smilingkylan — APPROVED** [review]

File: patches/@metamask+assets-controllers+12.0.0.patch (hunk 1 above)
Comment (kylanhurt) [reply patches/@metamask+assets-controllers+12.0.0.patch:2865]: @sethkfman

**Review by sethkfman — APPROVED** [review]
LGTM

**Pull request merged**